api/cwatch_api.php

```diff
@@ -287,65 +276,65 @@ public function getMalware($email, $domain)
-        $siteResponse = $this->getSite($email, $params['domain'], '', 'GET');
-        if (empty($siteResponse->errorMsg)) {
-            $site = json_decode($siteResponse->resp);
+        $siteResponse = $this->getSite($email, $params['domain']);
+        if (empty($siteResponse->errors())) {
+            $site = $siteResponse->response();
             $domainId = $site->engineSiteId;
         }
 
-        $response = $this->apiRequest('/domain/' . $domainId . '/settings/scanner/ftp', json_encode($params), 'POST');
-
-        return new CwatchResponse($response['content']);
+        return $this->apiRequest('/domain/' . $domainId . '/settings/scanner/ftp', $params, 'POST');
     }
 
     /**
      * Send an API request to cWatch server
      *
      * @param string $route The path to the API method
-     * @param string $body The data to be sent
+     * @param array $body The data to be sent
      * @param string $method Data transfer method (POST, GET, PUT, DELETE)
      * @return array
      */
-    private function apiRequest($route, $body, $method)
+    private function apiRequest($route, array $body, $method)
     {
         $url = $this->apiUrl . '/' . $route;
         $ch = curl_init();
 
-        curl_setopt($ch, CURLOPT_URL, $url);
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
-        curl_setopt($ch, CURLOPT_HEADER, 1);
-
         if ($method == 'POST') {
-            curl_setopt($ch, CURLOPT_POSTFIELDS, $body);
+            curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($body));
             curl_setopt($ch, CURLOPT_POST, 1);
         }
 
         if ($method == 'PUT') {
-            curl_setopt($ch, CURLOPT_POSTFIELDS, $body);
+            curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($body));
             curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'PUT');
         }
 
         if ($method == 'GET') {
+            $url .= empty($body) ? '' : '?' . urldecode(http_build_query($body));
             curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'GET');
         }
 
         if ($method == 'DELETE') {
-            curl_setopt($ch, CURLOPT_POSTFIELDS, $body);
+            curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($body));
             curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'DELETE');
         }
 
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+        curl_setopt($ch, CURLOPT_HEADER, 1);
+        curl_setopt($ch, CURLOPT_URL, $url);
+
         $headers = [];
         $headers[] = 'Authorization: ' . $this->token;
         $headers[] = 'Cache-Control: no-cache';
         $headers[] = 'Content-Type: application/json';
         curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
 
+        $this->lastRequest = ['content' => $body, 'headers' => $headers];
         $result = curl_exec($ch);
         if (curl_errno($ch)) {
-            throw new Exception('Curl Error: ' . curl_error($ch));
+            return new CwatchResponse('{"error": "Curl Error", "message": "' . curl_error($ch) . '", "status": 500}');
```

Comment:
```
$error = ['error' => 'Curl Error', 'message' => curl_error($ch), 'status' => 500];
return new CwatchResponse(json_encode($error));
```

Do we really want to show the cURL error though, or our own generic error instead? If users can see this in the interface it may not be good to display the curl error